partition-pruning.md

```diff
@@ -0,0 +1,259 @@
+# Partition pruning
+
+Partition pruning is a performance optimization only when the target table is a partitioned table. It analyzes the filter conditions in the query statements, selects only the partitions that may meet the conditions, and does not scan the partitions that do not match, thereby significantly reducing the amount of calculated data.
```

Comment:
- The optimization applies only to partitioned tables vs. the current language could imply it's always there and its an optimization when tables are partitioned/a negative in other cases.
- It avoids accessing the tables. There are other accesses besides scans - although TiDB calls what I would call a seek a scan, it is easier not to go into this detail here.
- Whether or not this is significant depends a lot on the query. If there are 10 partitions and pruning only eliminates 1 partition, it is unlikely to be significant.

```suggestion
Partition pruning is a performance optimization that applies to partitioned tables. It analyzes the filter conditions in query statements, and eliminates (_prunes_) partitions from consideration when they do not contain any data that will be required. By eliminating the non-required partitions, TiDB is able to reduce the amount of data that needs to be accessed and potentially significantly improving query execution times.
```